### PR TITLE
feat: Support Fairplay DRM in DASH manifest.

### PR DESCRIPTION
**Pull request by gmcgarry**
Support Fairplay DRM for DASH, specifying the license server in the manifest using dashif:Laurl.

For example, multi-drm manifest is here: https://content.media24.link/drm/manifest.mpd

### CODE REVIEW

**Comment by avelad** [comment]
Please, add tests of it on `dash_parser_content_protection_unit.js`. Thanks!

**Comment by avelad** [comment]
I've tried to manually test your PR with your content and it doesn't work. Do I need to make any other changes to make your manifest work? Thanks!

File: lib/dash/content_protection.js [hunk 1]
```diff
@@ -194,6 +194,27 @@ shaka.dash.ContentProtection = class {
...
 
+  /**
+   * Gets a FairPlay license URL from a content protection element
+   * containing a 'dashif:Laurl' element
+   *
+   * @param {shaka.dash.ContentProtection.Element} element
+   * @return {string}
+   */
+  static getFairPlayLicenseUrl(element) {
+    const dashIfLaurlNode = shaka.util.TXml.findChildNS(
```
[the remark this answers is not in the document]
Comment (avelad) [reply lib/dash/content_protection.js:205]: Note: You need add `DASH_MSE_ENCRYPTED_LEGACY_APPLE_MEDIA_KEYS_NOT_SUPPORTED`, see `HLS_MSE_ENCRYPTED_LEGACY_APPLE_MEDIA_KEYS_NOT_SUPPORTED` as example

**Comment by gmcgarry** [comment]
> I've tried to manually test your PR with your content and it doesn't work. Do I need to make any other changes to make your manifest work? Thanks!

It's probably the missing certificate.  Sorry about that.  Something like this:

`
player.configure({
  drm: {
    advanced: {
      'com.apple.fps': {
        serverCertificateUri: 'https://ott.dolby.com/OnDelKits/fairplay.cer'
      }
    }
  }
})
`

**Comment by avelad** [comment]
> > I've tried to manually test your PR with your content and it doesn't work. Do I need to make any other changes to make your manifest work? Thanks!
> 
> It's probably the missing certificate. Sorry about that. Something like this:
> 
> `player.configure({ drm: { advanced: { 'com.apple.fps': { serverCertificateUri: 'https://ott.dolby.com/OnDelKits/fairplay.cer' } } } })`

It seems to work, could we add this asset/stream to the Shaka Demo so that there is an example of this functionality?

File: lib/dash/content_protection.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (avelad) [reply lib/dash/content_protection.js:205]: @gmcgarry Please, add it!

**Comment by gmcgarry** [comment]
> > > I've tried to manually test your PR with your content and it doesn't work. Do I need to make any other changes to make your manifest work? Thanks!
> > 
> > 
> > It's probably the missing certificate. Sorry about that. Something like this:
> > `player.configure({ drm: { advanced: { 'com.apple.fps': { serverCertificateUri: 'https://ott.dolby.com/OnDelKits/fairplay.cer' } } } })`
> 
> It seems to work, could we add this asset/stream to the Shaka Demo so that there is an example of this functionality?

I can provide some better manifests.  But it's okay to use that manifest in the interim.

**Review by tykus160 — APPROVED** [review]

**Review by avelad — APPROVED** [review]

**Pull request merged**